### PR TITLE
[Localization] add missed move trigger and remove getPokemonMessage

### PR DESCRIPTION
**Pull request by EnochG1**
## What are the changes?
Add the option to localize message texts of some move-trigger which are not localized.
### list of messages & poke corpus message key (Just refer it, message can be different with og game.)
healHp : sv.btl_set.BTL_STRID_SET_HP_Recover)
sacrificialFullRestore (subject is different with pokerogue) : sv.btl_set.BTL_STRID_SET_IyasiNoNegai
invertStats : sv.btl_set.BTL_STRID_SET_Hikkurikaesu
resetStats : sv.btl_std.BTL_STRID_STD_RankReset
faintCountdown : sv.btl_std.BTL_STRID_STD_HorobiNoSango
copyType : swsh.btl_set.BTL_STRID_SET_MirrorType
suppressAbilities : sv.btl_set.BTL_STRID_SET_Ieki
swapArenaTags : sv.btl_set.BTL_STRID_SET_CourtChange

## Why am I doing these changes?
All text should be localizable.

## What did change?
 - Change hard-coded message for localize > src/data/move.ts
 - Added localization keys in each language folders. > move-trigger.ts
 - Edit EN copyType message following og game.
 - Remove redundant method > getPokemonMessage
 - Translated them to Korean. (+ each language after review)

#### En
<details>
<summary>Click to expand/collapse</summary>

| Case | Result |
| :-----: | :-----: |
| healHp | image |
| sacrificialFullRestore | image |
| invertStats | image |
| resetStats | image |
| faintCountdown | image |
| copyType | image |
| suppressAbilities | image |
| swapArenaTags | image |

</details>

#### It
<details>
<summary>Click to expand/collapse</summary>

| Case | Result |
| :-----: | :-----: |
| healHp | image |
| sacrificialFullRestore | image |
| invertStats | image |
| resetStats | image |
| faintCountdown | image |
| copyType | image |
| suppressAbilities | image |
| swapArenaTags | image |

</details>

#### Fr
<details>
<summary>Click to expand/collapse</summary>

| Case | Result |
| :-----: | :-----: |
| healHp | image |
| sacrificialFullRestore | image |
| invertStats | image |
| resetStats | image |
| faintCountdown | image |
| copyType | image |
| suppressAbilities | image |
| swapArenaTags | image |

</details>

#### De
<details>
<summary>Click to expand/collapse</summary>

| Case | Result |
| :-----: | :-----: |
| healHp | image |
| sacrificialFullRestore | image |
| invertStats | image |
| resetStats | image |
| faintCountdown | image |
| copyType | image |
| suppressAbilities | image |
| swapArenaTags | image |

</details>

#### Zh_CN
<details>
<summary>Click to expand/collapse</summary>

| Case | Result |
| :-----: | :-----: |
| healHp | image |
| sacrificialFullRestore | image |
| invertStats | image |
| resetStats | image |
| faintCountdown | image |
| copyType | image |
| suppressAbilities | image |
| swapArenaTags | image |

</details>

#### Zh_TW
<details>
<summary>Click to expand/collapse</summary>

| Case | Result |
| :-----: | :-----: |
| healHp | image |
| sacrificialFullRestore | image |
| invertStats | image |
| resetStats | image |
| faintCountdown | image |
| copyType | image |
| suppressAbilities | image |
| swapArenaTags | image |

</details>

#### Ko
<details>
<summary>Click to expand/collapse</summary>

| Case | Result |
| :-----: | :-----: |
| healHp | image |
| sacrificialFullRestore | image |
| invertStats | image |
| resetStats | image |
| faintCountdown | image |
| copyType | image |
| suppressAbilities | image |
| swapArenaTags | image |

</details>

## How to test the changes?
Manually (with src/overrides.ts)

### List of test-able moves
healHp
 - Moves.RECOVER, Moves.SOFT_BOILED, Moves.MILK_DRINK, Moves.SLACK_OFF, Moves.HEAL_PULSE...

sacrificialFullRestore
 - Moves.HEALING_WISH

invertStats
 - Moves.TOPSY_TURVY

resetStats
 - Moves.HAZE, Moves.CLEAR_SMOG, Moves.FREEZY_FROST

faintCountdown
 - Moves.PERISH_SONG

copyType
 - Moves.REFLECT_TYPE

suppressAbilities
 - Moves.GASTRO_ACID

swapArenaTags
 - Moves.COURT_CHANGE

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [ ] Have I considered writing automated tests for the issue?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/locales/zh_TW/move-trigger.ts [hunk 1]
```diff
@@ -50,5 +50,13 @@ export const moveTriggers: SimpleTranslationEntries = {
...
-  "cannotUseMove": "{{pokemonName}}\n無法使用{{moveName}}！"
+  "cannotUseMove": "{{pokemonName}}\n無法使用{{moveName}}！",
+  "healHp": "{{pokemonName}} had its HP restored.",
+  "sacrificialFullRestore": "{{pokemonName}}'s Healing Wish\nwas granted!",
+  "invertStats": "{{pokemonName}}'s stat changes\nwere all reversed!",
+  "resetStats": "{{pokemonName}}'s stat changes\nwere eliminated!",
+  "faintCountdown": "{{pokemonName}}\nwill faint in {{turnCount}} turns.",
+  "copyType": "{{pokemonName}}'s type\nchanged to match {{targetPokemonName}}'s!",
+  "suppressAbilities": "{{pokemonName}}'s ability\nwas suppressed!",
+  "swapArenaTags": "{{pokemonName}} swapped the battle effects affecting each side of the field!",
```
Comment (mercurius-00) [inline src/locales/zh_TW/move-trigger.ts:61]:
```suggestion
  "healHp": "{{pokemonName}}的\n體力回復了！",
  "sacrificialFullRestore": "{{pokemonName}}的\n治癒之願實現了！",
  "invertStats": "{{pokemonName}}的\n能力變化顛倒過來了！",
  "resetStats": "{{pokemonName}}的\n能力變化復原了！",
  "faintCountdown": "{{pokemonName}}\n將在{{turnCount}}回合後滅亡！",
  "copyType": "{{pokemonName}}變成了{{targetPokemonName}}的屬性！",
  "suppressAbilities": "{{pokemonName}}的特性\n變得無效了！",
  "swapArenaTags": "{{pokemonName}}\n交換了雙方的場地效果！
",
```

File: src/locales/zh_CN/move-trigger.ts [hunk 2]
```diff
@@ -50,5 +50,13 @@ export const moveTriggers: SimpleTranslationEntries = {
...
-  "cannotUseMove": "{{pokemonName}}\n无法使用{{moveName}}！"
+  "cannotUseMove": "{{pokemonName}}\n无法使用{{moveName}}！",
+  "healHp": "{{pokemonName}} had its HP restored.",
+  "sacrificialFullRestore": "{{pokemonName}}'s Healing Wish\nwas granted!",
+  "invertStats": "{{pokemonName}}'s stat changes\nwere all reversed!",
+  "resetStats": "{{pokemonName}}'s stat changes\nwere eliminated!",
+  "faintCountdown": "{{pokemonName}}\nwill faint in {{turnCount}} turns.",
+  "copyType": "{{pokemonName}}'s type\nchanged to match {{targetPokemonName}}'s!",
+  "suppressAbilities": "{{pokemonName}}'s ability\nwas suppressed!",
+  "swapArenaTags": "{{pokemonName}} swapped the battle effects affecting each side of the field!",
```
Comment (mercurius-00) [inline src/locales/zh_CN/move-trigger.ts:61]:
```suggestion
  "healHp": "{{pokemonName}}的\n体力回复了！",
  "sacrificialFullRestore": "{{pokemonName}}的\n治愈之愿实现了！",
  "invertStats": "{{pokemonName}}的\n能力变化颠倒过来了！",
  "resetStats": "{{pokemonName}}的\n能力变化复原了！",
  "faintCountdown": "{{pokemonName}}\n将在{{turnCount}}回合后灭亡！",
  "copyType": "{{pokemonName}}\n变成了{{targetPokemonName}}的属性！",
  "suppressAbilities": "{{pokemonName}}的特性\n变得无效了！",
  "swapArenaTags": "{{pokemonName}}\n交换了双方的场地效果！",
```

**Review by mercurius-00 — COMMENTED** [review]
zh-CN and zh-TW

File: src/locales/zh_TW/move-trigger.ts (hunk 1 above)
Comment (EnochG1) [reply src/locales/zh_TW/move-trigger.ts:61]: Please check last line

File: src/data/move.ts [hunk 3]
```diff
@@ -1482,7 +1482,7 @@ export class SacrificialFullRestoreAttr extends SacrificialAttr {
     const maxPartyMemberHp = user.scene.getParty().map(p => p.getMaxHp()).reduce((maxHp: integer, hp: integer) => Math.max(hp, maxHp), 0);
 
     user.scene.pushPhase(new PokemonHealPhase(user.scene, user.getBattlerIndex(),
-      maxPartyMemberHp, getPokemonMessage(user, "'s Healing Wish\nwas granted!"), true, false, false, true), true);
+      maxPartyMemberHp, i18next.t("moveTriggers:sacrificialFullRestore", {pokemonName: getPokemonNameWithAffix(user)}), true, false, false, true), true);
```
Comment (CodeTappert) [inline src/data/move.ts:1485]:
```suggestion
      maxPartyMemberHp, i18next.t("moveTriggers:sacrificialFullRestore", {pokemonName: getPokemonNameWithAffix(user), target: getPokemonNameWithAffix(target) }), true, false, false, true), true);
```

File: src/locales/zh_TW/move-trigger.ts [hunk 4]
```diff
@@ -50,5 +50,13 @@ export const moveTriggers: SimpleTranslationEntries = {
   "transformedIntoTarget": "{{pokemonName}}\n變身成了{{targetName}}！",
   "tryingToTakeFoeDown": "{{pokemonName}}\n想和對手同歸於盡！",
   "addType": "{{pokemonName}}\n增加了{{typeName}}屬性！",
-  "cannotUseMove": "{{pokemonName}}\n無法使用{{moveName}}！"
+  "cannotUseMove": "{{pokemonName}}\n無法使用{{moveName}}！",
+  "healHp": "{{pokemonName}}的\n體力回復了！",
+  "sacrificialFullRestore": "{{pokemonName}}的\n治癒之願實現了！",
```
Comment (mercurius-00) [inline src/locales/zh_TW/move-trigger.ts:55]:
```suggestion
  "sacrificialFullRestore": "治癒之願\n在{{target}}身上實現了！",
```

File: src/locales/zh_CN/move-trigger.ts [hunk 5]
```diff
@@ -50,5 +50,13 @@ export const moveTriggers: SimpleTranslationEntries = {
   "transformedIntoTarget": "{{pokemonName}}\n变身成了{{targetName}}！",
   "tryingToTakeFoeDown": "{{pokemonName}}\n想和对手同归于尽！",
   "addType": "{{pokemonName}}\n增加了{{typeName}}属性！",
-  "cannotUseMove": "{{pokemonName}}\n无法使用{{moveName}}！"
+  "cannotUseMove": "{{pokemonName}}\n无法使用{{moveName}}！",
+  "healHp": "{{pokemonName}}的\n体力回复了！",
+  "sacrificialFullRestore": "{{pokemonName}}的\n治愈之愿实现了！",
```
Comment (mercurius-00) [inline src/locales/zh_CN/move-trigger.ts:55]:
```suggestion
  "sacrificialFullRestore": "治愈之愿\n在{{target}}身上实现了！",
```

**Review by mercurius-00 — COMMENTED** [review]
(zh-CN, zh-TW)
change sacrificialFullRestore to match original game text

File: src/locales/de/move-trigger.ts [hunk 6]
```diff
@@ -50,5 +50,13 @@ export const moveTriggers: SimpleTranslationEntries = {
   "transformedIntoTarget": "{{pokemonName}} verwandelt sich in {{targetName}}!",
   "tryingToTakeFoeDown": "{{pokemonName}} versucht, den Angreifer mit sich zu nehmen!",
   "addType": "{{pokemonName}} nimmt zusätzlich den Typ {{typeName}} an!",
-  "cannotUseMove": "{{pokemonName}} kann {{moveName}} nicht einsetzen!"
+  "cannotUseMove": "{{pokemonName}} kann {{moveName}} nicht einsetzen!",
+  "healHp": "KP von {{pokemonName}} wurden aufgefrischt!",
+  "sacrificialFullRestore": "Das Heilopfer erreicht {{target}}",
+  "invertStats": "Alle Statusveränderungen von {{pokemonName}} wurden invertiert!",
+  "resetStats": "Die Statusveränderungen von {{pokemonName}} wurden aufgehoben!!",
+  "faintCountdown": "{{pokemonName}} geht nach {{turnCount}} Runden K.O.!",
```
Comment (mercurius-00) [inline src/locales/de/move-trigger.ts:58]:
```suggestion
  "faintCountdown": "{{pokemonName}} geht nach {{turnCount}} Runden K.O.!",
```

**Review by mercurius-00 — COMMENTED** [review]
delete trailing space

File: src/locales/fr/move-trigger.ts [hunk 7]
```diff
@@ -50,5 +50,13 @@ export const moveTriggers: SimpleTranslationEntries = {
...
-  "cannotUseMove": "{{pokemonName}} ne peut pas\nutiliser la capacité {{moveName}} !"
+  "cannotUseMove": "{{pokemonName}} ne peut pas\nutiliser la capacité {{moveName}} !",
+  "healHp": "{{pokemonName}} had its HP restored.",
+  "sacrificialFullRestore": "{{pokemonName}}'s Healing Wish\nwas granted!",
+  "invertStats": "{{pokemonName}}'s stat changes\nwere all reversed!",
+  "resetStats": "{{pokemonName}}'s stat changes\nwere eliminated!",
+  "faintCountdown": "{{pokemonName}}\nwill faint in {{turnCount}} turns.",
+  "copyType": "{{pokemonName}}'s type\nchanged to match {{targetPokemonName}}'s!",
+  "suppressAbilities": "{{pokemonName}}'s ability\nwas suppressed!",
+  "swapArenaTags": "{{pokemonName}} swapped the battle effects affecting each side of the field!",
```
Comment (Adri1) [inline src/locales/fr/move-trigger.ts:61]:
```suggestion
  "healHp": "{{pokemonName}}\nrécupère des PV !",
  "sacrificialFullRestore": "Le Vœu Soin est exaucé et profite\nà {{pokemonName}} !",
  "invertStats": "Les changements de stats\nde {{pokemonName}} sont inversés !",
  "resetStats": "Les changements de stats\nde {{pokemonName}} ont tous été annulés !",
  "faintCountdown": "{{pokemonName}}\nsera K.O. dans {{turnCount}} tours !",
  "copyType": "{{pokemonName}} prend le type\nde {{targetPokemonName}} !",
  "suppressAbilities": "Le talent de {{pokemonName}}\na été rendu inactif !",
  "swapArenaTags": "Les effets affectant chaque côté du terrain\nont été échangés par {{pokemonName}} !",
```

**Review by Adri1 — COMMENTED** [review]
Added French

File: src/locales/it/move-trigger.ts [hunk 8]
```diff
@@ -50,5 +50,13 @@ export const moveTriggers: SimpleTranslationEntries = {
...
-  "cannotUseMove": "{{pokemonName}} non può usare {{moveName}}!"
+  "cannotUseMove": "{{pokemonName}} non può usare {{moveName}}!",
+  "healHp": "{{pokemonName}} had its HP restored.",
+  "sacrificialFullRestore": "{{pokemonName}}'s Healing Wish\nwas granted!",
+  "invertStats": "{{pokemonName}}'s stat changes\nwere all reversed!",
+  "resetStats": "{{pokemonName}}'s stat changes\nwere eliminated!",
+  "faintCountdown": "{{pokemonName}}\nwill faint in {{turnCount}} turns.",
+  "copyType": "{{pokemonName}}'s type\nchanged to match {{targetPokemonName}}'s!",
+  "suppressAbilities": "{{pokemonName}}'s ability\nwas suppressed!",
+  "swapArenaTags": "{{pokemonName}} swapped the battle effects affecting each side of the field!",
```
Comment (NicusPulcis) [inline src/locales/it/move-trigger.ts:61]:
```suggestion
  "healHp": "{{pokemonName}} ha recuperato dei PS.",
  "sacrificialFullRestore": "{{pokemonName} riceve i benefici\neffetti di Curardore!",
  "invertStats": "Le modifiche alle statistiche di {{pokemonName}}\nvengono invertite!",
  "resetStats": "Tutte le modifiche alle statistiche sono state annullate!",
  "faintCountdown": "{{pokemonName}}\nandrà KO dopo {{turnCount}} turni.",
  "copyType": "{{pokemonName}} assume il tipo\ndi {{targetPokemonName}}!",
  "suppressAbilities": "L’abilità di {{pokemonName}}\nperde ogni efficacia!",
  "swapArenaTags": "{{pokemonName}} ha invertito gli effetti attivi\nnelle due metà del campo!",
```

**Review by NicusPulcis — APPROVED** [review]
italian localization

**Comment by EnochG1** [comment]
@CodeTappert 
I reverted your suggestion of move.ts because target is null.
I realized the timing of set message is when pokemon used healing wish, not when changed pokemon go out.

image

So please modify your message to match the subject.

**Review by mercurius-00 — APPROVED** [review]

File: src/locales/de/move-trigger.ts [hunk 9]
```diff
@@ -50,5 +50,13 @@ export const moveTriggers: SimpleTranslationEntries = {
   "transformedIntoTarget": "{{pokemonName}} verwandelt sich in {{targetName}}!",
   "tryingToTakeFoeDown": "{{pokemonName}} versucht, den Angreifer mit sich zu nehmen!",
   "addType": "{{pokemonName}} nimmt zusätzlich den Typ {{typeName}} an!",
-  "cannotUseMove": "{{pokemonName}} kann {{moveName}} nicht einsetzen!"
+  "cannotUseMove": "{{pokemonName}} kann {{moveName}} nicht einsetzen!",
+  "healHp": "KP von {{pokemonName}} wurden aufgefrischt!",
+  "sacrificialFullRestore": "Das Heilopfer erreicht {{target}}",
```
Comment (CodeTappert) [inline src/locales/de/move-trigger.ts:55]:
```suggestion
  "sacrificialFullRestore": "Das Heilopfer von {{pokemonName}} erreicht sein Ziel!",
```

File: src/locales/de/move-trigger.ts [hunk 10]
```diff
@@ -50,5 +50,13 @@ export const moveTriggers: SimpleTranslationEntries = {
   "transformedIntoTarget": "{{pokemonName}} verwandelt sich in {{targetName}}!",
   "tryingToTakeFoeDown": "{{pokemonName}} versucht, den Angreifer mit sich zu nehmen!",
   "addType": "{{pokemonName}} nimmt zusätzlich den Typ {{typeName}} an!",
-  "cannotUseMove": "{{pokemonName}} kann {{moveName}} nicht einsetzen!"
+  "cannotUseMove": "{{pokemonName}} kann {{moveName}} nicht einsetzen!",
+  "healHp": "KP von {{pokemonName}} wurden aufgefrischt!",
+  "sacrificialFullRestore": "Das Heilopfer von {{pokemonName}} erreicht sein Ziel!",
+  "invertStats": "Alle Statusveränderungen von {{pokemonName}} wurden invertiert!",
+  "resetStats": "Die Statusveränderungen von {{pokemonName}} wurden aufgehoben!!",
```
Comment (CodeTappert) [inline src/locales/de/move-trigger.ts:57]:
```suggestion
  "resetStats": "Die Statusveränderungen von {{pokemonName}} wurden aufgehoben!",
```

**Review by CodeTappert — APPROVED** [review]

**Review by CodeTappert — APPROVED** [review]

**Review by Adri1 — APPROVED** [review]

File: src/locales/pt_BR/move-trigger.ts [hunk 11]
```diff
@@ -50,5 +50,13 @@ export const moveTriggers: SimpleTranslationEntries = {
...
-  "cannotUseMove": "{{pokemonName}} não pode usar {{moveName}}!"
+  "cannotUseMove": "{{pokemonName}} não pode usar {{moveName}}!",
+  "healHp": "{{pokemonName}} had its HP restored.",
+  "sacrificialFullRestore": "{{pokemonName}}'s Healing Wish\nwas granted!",
+  "invertStats": "{{pokemonName}}'s stat changes\nwere all reversed!",
+  "resetStats": "{{pokemonName}}'s stat changes\nwere eliminated!",
+  "faintCountdown": "{{pokemonName}}\nwill faint in {{turnCount}} turns.",
+  "copyType": "{{pokemonName}}'s type\nchanged to match {{targetPokemonName}}'s!",
+  "suppressAbilities": "{{pokemonName}}'s ability\nwas suppressed!",
+  "swapArenaTags": "{{pokemonName}} swapped the battle effects affecting each side of the field!",
```
Comment (josericardo-fo) [inline src/locales/pt_BR/move-trigger.ts:61]:
```suggestion
  "cannotUseMove": "{{pokemonName}} não pode usar {{moveName}}!",
  "healHp": "{{pokemonName}} teve seus PS recuperados.",
  "sacrificialFullRestore": "O Healing Wish de {{pokemonName}}\nfoi concedido!",
  "invertStats": "As mudanças de atributo de {{pokemonName}}\nforam revertidas!",
  "resetStats": "As mudanças de atributo de {{pokemonName}}\nforam eliminadas!",
  "faintCountdown": "{{pokemonName}}\nirá desmaiar em {{turnCount}} turnos.",
  "copyType": "O tipo de {{pokemonName}}\nmudou para combinar com {{targetPokemonName}}!",
  "suppressAbilities": "A habilidade de {{pokemonName}}\nfoi suprimida!",
  "swapArenaTags": "{{pokemonName}} trocou os efeitos de batalha que afetam cada lado do campo!",
```

**Pull request merged**